Elevator/ElevatorControlPanel.lua

```diff
@@ -14,37 +14,144 @@ function drawButton(scr)
+				drawButton(createButton(i,colors.white, "")) -- "None" spot
+				allowed[i] = false
+			end
+		end
+	end
+	while true do -- Wait till you get a valid input
+		_,_,_,y = os.pullEvent("monitor_touch")
+		if y and allowed[y] then return y end
+	end
+end
+
+function ConsoleEvent() -- Enter computer to get console
+	mon.clear()
+	print("Operations:\n1. Add floor\n2. Edit floor\n3.Delete floor\nPlease pick one (1-3)...")
+	operation = string.sub(read(), 1,1) -- Limit to first character
+	operation = tonumber(operation)
+	if operation then -- Is a number
+		if operation == 1 then -- Add new floor
+			local color = ColorPicker() -- Pick a color
+			mon.clear()
+			local floor = FloorPicker(false) -- Pick a free floor position
+			mon.clear()
+			mon.setCursorPos(1, 2)
+			print("Name: ")
+			local name = read()
+			floors[floor] = createButton(floor,color, name)
+			
+		elseif operation == 2 then -- Edit floor
+			local color = ColorPicker() -- Pick a color
+			mon.clear()
+			local floor = FloorPicker(true)  -- Pick an existing floor position
+			mon.clear()
+			mon.setCursorPos(1, 2)
+			print("Name: ")
+			local name = read()
+			floors[floor] = createButton(floor,color, name) -- Overwrite with new parameters
+			
+		elseif operation == 3 then -- Delete floor
+			local floor = FloorPicker(true) -- Pick an existing floor position
+			floors[floor] = nil -- "Delete" floor position
+		else -- No valid input
+			mon.write("Invalid!")
+		end
+		drawFloors()
+	end
+end
+
 --use advanced monitor on top for rendering
 mon = peripheral.wrap("top")
 term.redirect(mon)
+--smaller text = more pixels to work with
+mon.setTextScale(0.5)
+mon.clear()
 
 --last button pressed
 local lastButtonHit = 0
 --list of buttons
-local floors = {}
-floors[0] = createButton(2, colors.orange, "button 0")
-floors[1] = createButton(3, colors.cyan, "button 1")
-floors[2] = createButton(4, colors.red, "button 2")
+floors = {}
+floors[1] = createButton(1, colors.orange, "Preset Floor 1")
+floors[2] = createButton(2, colors.cyan, "Preset Floor 2")
+floors[3] = createButton(3, colors.red, "Preset Floor 3")
+drawFloors(floors)
 
---smaller text = more pixels to work with
-mon.setTextScale(0.5)
-mon.clear()
 while true do
-  _, _, cx, cy = os.pullEvent("monitor_touch")
-  --for each line on screen
-  for i = 0, 10 do
-    --only process defined buttons
-    if(floors[i]) then
-      --for 1 wide buttons this is ok - if y touched equals the button's y then we count it as pressed
-      if(cy == floors[i].y) then
-        --add chevrons to whichever button was hit and remove from previous
-        floors[lastButtonHit].text = floors[lastButtonHit].text:gsub(">>","")
-        drawButton(floors[lastButtonHit])
-        floors[i].text = ">>"..floors[i].text
-        lastButtonHit = i
-      end
-      --render
-      drawButton(floors[i])
-    end
+  local eventData = {os.pullEvent()} -- Pull all events
+  
+  if eventData[1] == "monitor_touch" then
```

Comment:
Might want to create some kind of enums with indexes of different event returns, like event_type=1, keycode=2 and so on

Reply:
Yes, event handler module would be nice